### ISSUE TITLE
Units Errors: print constraint causing the error?

### ISSUE THREAD

**Issue by cjk12** [hoburg/gpkit#657]
So I'm getting this error:

```
ValueError                                Traceback (most recent call last)
<ipython-input-23-1a920353b2ff> in <module>()
    100 
    101 # m.controlpanel(fn_of_sol=Weights_pie)
--> 102 sol=m.localsolve()
    103 # sol=m.solve()

/Users/codykarcher/Dropbox (MIT)/Research/gpkit/gpkit/constraints/prog_factories.pyc in solvefn(self, solver, verbosity, skipsweepfailures, *args, **kwargs)
     78         else:
     79             self.program, solvefn = genfunction(self, verbosity-1)
---> 80             result = solvefn(solver, verbosity-1, *args, **kwargs)
     81             solution.append(result)
     82         solution.program = self.program

/Users/codykarcher/Dropbox (MIT)/Research/gpkit/gpkit/constraints/signomial_program.pyc in localsolve(self, solver, verbosity, x0, rel_tol, iteration_limit, **kwargs)
    108                 result = gp.solve(solver, verbosity-1, **kwargs)
    109             except (RuntimeWarning, ValueError):
--> 110                 nearest_feasible = feasibility_model(gp, "max")
    111                 self.gps.append(nearest_feasible)
    112                 result = nearest_feasible.solve(verbosity=verbosity-1)

/Users/codykarcher/Dropbox (MIT)/Research/gpkit/gpkit/feasibility.pyc in feasibility_model(program, flavour, varname, constants, signomials, programType)
     49         posynomials = ([slackvar >= 1] +
     50                        [posy <= slackvar
---> 51                         for posy in posynomials[1:]])
     52         prog = programType(slackvar, posynomials)
     53 

/Users/codykarcher/Dropbox (MIT)/Research/gpkit/gpkit/nomials/nomial_math.pyc in __ge__(self, other)
    412     def __ge__(self, other):
    413         if isinstance(other, Numbers + (Posynomial,)):
--> 414             return PosynomialInequality(self, ">=", other)
    415         else:
    416             # fall back on other's __ge__

/Users/codykarcher/Dropbox (MIT)/Research/gpkit/gpkit/nomials/nomial_math.pyc in __init__(self, left, oper, right)
    464         self.substitutions = dict(p_lt.values)
    465         self.substitutions.update(m_gt.values)
--> 466         self._unsubbed = self._gen_unsubbed()
    467         self.nomials = [self.left, self.right, self.p_lt, self.m_gt]
    468         self.nomials.extend(self._unsubbed)

/Users/codykarcher/Dropbox (MIT)/Research/gpkit/gpkit/nomials/nomial_math.pyc in _gen_unsubbed(self)
    506                                  " on both sides: '%s' and '%s' can not"
    507                                  " be converted into each other."
--> 508                                  "" % (self.p_lt.units, self.m_gt.units))
    509 
    510         p.exps, p.cs = self._simplify_posy_ineq(p.exps, p.cs)

ValueError: constraints must have the same units on both sides: '1 meter ** 4 / foot / kilogram' and 'None' can not be converted into each other.
```

But what I really want is for it to tell me what constraint is throwing me the error.  I'm spending way to much time hunting down units errors...

**Comment by cjk12** [comment]
I'll post the code when I can find the error and make a MWE.

**Comment by whoburg** [comment]
@cjk12 we just merged #670; unit mismatches will now print the offending terms.
